internal/pkg/service/stream/api/mapper/sink.go

```diff
@@ -9,14 +9,23 @@ import (
 )
 
 func (m *Mapper) NewSinkStatisticsTotalResponse(result statistics.Aggregated) *stream.SinkStatisticsTotalResult {
-	return &stream.SinkStatisticsTotalResult{
+	res := &stream.SinkStatisticsTotalResult{
 		Total: mapValueToLevel(result.Total),
 		Levels: &stream.Levels{
 			Local:   mapValueToLevel(result.Local),
 			Staging: mapValueToLevel(result.Staging),
 			Target:  mapValueToLevel(result.Target),
 		},
 	}
+
+	if res.Total == nil {
```

Comment:
Cannot we use some other function `mapValueToLevel` that returns the `stream.Level` instead of `*stream.Level`? It prevents reasigning it into empty value.